### ISSUE TITLE
`TFLiteConverter` breaks `SignatureDefs` when used with full integer quantization

### ISSUE THREAD

**Issue by lgeiger** [tensorflow/tensorflow#48148]
**System information**

- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): macOS
- TensorFlow installation (pip package or built from source): binary 
- TensorFlow library (version, if pip package or github SHA, if built from source): `2.5.0-dev20210319` and `2.6.0-dev20210329`

**Describe the current behavior**

Converting `SignatureDefs` together with full int8 quantization by setting
```python
converter.inference_input_type = tf.int8
converter.inference_output_type = tf.int8
```
breaks the signature runner since the signature still refers to tensors that [have been removed when converting to full int8 quantization](https://github.com/tensorflow/tensorflow/blob/305926798c36e2babfd31395317db33c55490d5b/tensorflow/lite/python/lite.py#L745).

Please see [this notebook](https://colab.research.google.com/drive/1stwy8M-tLDBCPW8_BRXpBs5ZOzlm94Jz?usp=sharing) for the full code to reproduce the problem.

For more background on why this is problematic for multi-output models please checkout #47927

**Comment by abattery** [comment]
Thanks again @lgeiger for reporting this!

**Comment by karimnosseir** [comment]
Thanks @lgeiger for reporting this. We will work on a fix.

Thanks

**Comment by lgeiger** [comment]
This has been fixed in a0e5ec8cb940fbea193706f0c6609be6dad5a961 by @MeghnaNatraj so the issue doesn't exist in tf-nightly anymore.
